extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php

```diff
@@ -15,6 +15,159 @@ class SEOTweaksHooksHelper {
+        $imageServing = new ImageServing( [ $title->getArticleID() ], $width, $height );
+        $out = $imageServing->getImages( 1, $driverName);
+        return self::createTitleFromResultArray( $out );
+    }
+
+    /**
+     * @desc Creates a Title object from array of file names
+     * @param $out array of file names
+     * @return null|\Title
+     */
+    static protected function createTitleFromResultArray( $out ) {
+        if ( !empty( $out ) ) {
+            ///used reset instead direct call because we can get hashmap from ImageServing driver.
+            $first = reset( $out );
+            $name = $first[ 0 ][ 'name' ];
+            return Title::newFromText( $name, NS_FILE );
+        }
+        return null;
+    }
+
+    /**
+     * @param $meta
+     * @param $title Title
+     * @return bool
+     */
+    static public function onOpenGraphMetaHeaders( &$meta, $title ) {
+        global $wgMemc;
+        if ( !empty( $title ) && $title instanceof Title && !$title->isMainPage() ) {
+            $namespace = $title->getNamespace();
+            if ( $namespace == NS_USER ) {
+                return true;
+            }
+            $cacheKey = self::makeOpenGraphKey( $title );
+            $imageUrl = $wgMemc->get( $cacheKey );
+
+            //if no image in memcache
+            if ( empty( $imageUrl ) ) {
+                if ( $namespace != NS_FILE ) {
+                    $title = self::getFirstArticleImage( $title );
+                }
+                if ( !empty( $title ) ) {
+                    $file = wfFindFile( $title );
+                    if ( !empty( $file ) ) {
+                        $thumb = self::getResizeImageUrlIfLargerThanMax( $file );
+                        if ( !empty( $thumb ) ) $meta[ "og:image" ] = $thumb;
+                    }
+                }
+                if ( isset( $meta[ "og:image" ] ) && ( !empty( $meta[ "og:image" ] ) ) ) {
+                    $imageUrl = $meta[ "og:image" ];
+                } else {
+                    // Even if there is no og:image, we store the info in memcahe so we don't do the
+                    // processing again
+                    $imageUrl = '';
+                }
+                $wgMemc->set( $cacheKey, $imageUrl );
+            }
+
+            // only when there is a thumbnail url add it to metatags
+            if ( !empty( $imageUrl ) ) {
+                $meta[ 'og:image' ] = $imageUrl;
+            }
+        }
+
+        return true;
+    }
+
+    static public function onArticleRobotPolicy( &$policy, Title $title ) {
+
+        $ns = MWNamespace::getSubject( $title->getNamespace() );
+
+        if ( in_array( $ns, [ NS_MEDIAWIKI, NS_TEMPLATE ] ) ) {
+            $policy = [
+                'index' => 'noindex',
+                'follow' => 'follow'
+            ];
+        }
+        return true;
+    }
+
+    static public function onShowMissingArticle( $article ) {
+        global $wgOut;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)